src/Uno.UI.Composition/Composition/Uno/DrawingFilter.skia.cs

```diff
@@ -13,15 +14,21 @@ namespace Uno.UI.Composition;
 ///		The opacity to apply to children.
 ///		This should be applied only on drawing and not the container to avoid fade-ou of child containers.
 /// </param>
-internal record struct DrawingFilters(float Opacity)
+internal readonly record struct DrawingFilters(float Opacity)
 {
 	public static DrawingFilters Default { get; } = new(1.0f);
 
-	private SKColorFilter? _opacityColorFilter = null;
-
 	// Note: This SKColorFilter might be created more than once since this Filter is a struct.
 	//       However since this Filter is copied (pushed to the stack) only when something changes, it should still catch most cases.
 	public SKColorFilter? OpacityColorFilter => Opacity is 1.0f
 		? null
-		: _opacityColorFilter ??= SKColorFilter.CreateBlendMode(new SKColor(0xFF, 0xFF, 0xFF, (byte)(255 * Opacity)), SKBlendMode.Modulate);
+		: _opacityToColorFilter.Invoke((byte)(255 * Opacity));
+
+	// only 255 possible values
+	private static readonly Func<byte, SKColorFilter> _opacityToColorFilter = FuncMemoizeExtensions.AsMemoized((byte opacity)
```

Comment:
I'm not sure if it's worth doing it.
Memoizing still has its own costs as well as it causes a dictionary lookup. The benefit isn't so clear to me.

Reply:
I refactored it to use a static array of size 255.